### ISSUE TITLE
User with DEVELOPER role are unable to retrieve environment's variables

### ISSUE THREAD

**Issue by vincenzodnp** [uselagoon/lagoon-cli#176]
**Describe the bug**
A user with DEVELOPER role, is not able to retrieve development environment's variable.
Indeed the user gets `Error: graphql: Unauthorized: You don't have permission to "environment:view:production" on "env_var".`
Here is the `--debug` output of the command:
```
lagoon list variables -p drupal-example  -e develop --debug
Request: {"query":"\nquery ($name: String!){\nproject:projectByName(name: $name) {\n  ...Project\n}\n}fragment Project on Project {\nid\nname\nenvironments {\nopenshiftProjectName\nname\nenvVariables {\nid\nname\nscope\n}\n}\n}","variables":[{"name":"name","value":"drupal-example"},{"name":"openshift","value":{}}]}
Error: graphql: Unauthorized: You don't have permission to "environment:view:production" on "env_var". 
```
Here the same command but from a user with higher permissions 
```
lagoon list variables -p drupal-example  -e develop --debug
Request: {"query":"\nquery ($name: String!){\nproject:projectByName(name: $name) {\n  ...Project\n}\n}fragment Project on Project {\nid\nname\nenvironments {\nopenshiftProjectName\nname\nenvVariables {\nid\nname\nscope\n}\n}\n}","variables":[{"name":"openshift","value":{}},{"name":"name","value":"drupal-example"}]} 

Response: {"environments":[{"envVariables":[{"id":xxxxx,"name":"BASIC_AUTH_USERNAME","scope":"runtime"},{"id":xxxxx,"name":"BASIC_AUTH_PASSWORD","scope":"runtime"},{"id":xxxxx,"name":"DOCKER_REGISTRY","scope":"global"},{"id":xxxxx,"name":"DOCKER_CLI_IMAGE_URI","scope":"global"},{"id":xxxxx,"name":"DOCKER_PHP_IMAGE_URI","scope":"global"},{"id":xxxxx,"name":"DOCKER_NGINX_IMAGE_URI","scope":"global"}],"name":"develop","openshiftProjectName":"drupal-example-develop"},{"envVariables":[{"id":xxxxx,"name":"BASIC_AUTH_USERNAME","scope":"runtime"},{"id":xxxxx,"name":"BASIC_AUTH_PASSWORD","scope":"runtime"}],"name":"master","openshiftProjectName":"drupal-example-master"},{"envVariables":[{"id":xxxxx,"name":"BASIC_AUTH_USERNAME","scope":"runtime"},{"id":xxxxx,"name":"BASIC_AUTH_PASSWORD","scope":"runtime"}],"name":"qa","openshiftProjectName":"drupal-example-qa"}],"id":xxxxx,"name":"drupal-example"}

Request: {"query":"\nquery ($name: String!, $project: Int!) {\nenvironmentByName(name: $name, project: $project) {\n...Environment\n}\n}fragment Environment on Environment {\nopenshiftProjectName\nname\nenvVariables {\nid\nname\nscope\n}\n}","variables":[{"name":"name","value":"develop"},{"name":"project","value":xxxxx}]}

Response: {"envVariables":[{"id":xxxxx,"name":"BASIC_AUTH_USERNAME","scope":"runtime"},{"id":xxxxx,"name":"BASIC_AUTH_PASSWORD","scope":"runtime"},{"id":xxxxx,"name":"DOCKER_REGISTRY","scope":"global"},{"id":xxxxx,"name":"DOCKER_CLI_IMAGE_URI","scope":"global"},{"id":xxxxx,"name":"DOCKER_PHP_IMAGE_URI","scope":"global"},{"id":xxxxx,"name":"DOCKER_NGINX_IMAGE_URI","scope":"global"}],"name":"develop","openshiftProjectName":"drupal-example-develop"}
```
So it seems the lagoon cli runs two graphql queries: `projectByName` and `environmentByName`.
The first query returns the variables for all the environments (production included) and that's why the command run by a user with DEVELOPER role breaks.

**To Reproduce**
1. Run `lagoon list variables -p drupal-example  -e develop --debug` as user with Developer role
2. See error

**Expected behavior**
Retrieve the variables ONLY for the selected environment

**Comment by shreddedbacon** [comment]
The query that is executed returns an error for developer because the lagoon-cli doesn't know how to determine if the user has permission to `environmen:view:production`.

```
query ($name: String!){
	project:projectByName(name: $name) {
	  	...Project
	}
}
fragment Project on Project {
	id
	name
	environments {
		openshiftProjectName
		name
		envVariables {
			id
			name
			scope
		}
	}
}
```

If the user is a DEVELOPER role, then we would need to use this fragment instead.
```
fragment Project on Project {
	id
	name
	environments(type:DEVELOPMENT) {
		openshiftProjectName
		name
		envVariables {
			id
			name
			scope
		}
	}
}
```

When you run the query as a developer using GraphQL directly, it will probably return an error, but it will also probably return usable data.

With the lagoon-cli, when it sees an error, it exits straight away so you don't even get to see if there was usable data.

I'm not sure how best to approach solving this in the lagoon-cli though.

**Comment by rocketeerbkw** [comment]
The cli in this case is over querying since the user will only want to list env vars for a single environment at a time. We should change the query to use `environmentByName` or something similar.

**Comment by shreddedbacon** [comment]
https://github.com/uselagoon/lagoon/pull/3204 introduced a new query resolver that introduced [this input](https://github.com/uselagoon/lagoon/blob/7f5705049c73d4127eab2d498a67fc80777b5d56/services/api/src/typeDefs.js#L1039-L1042) that allows for 
 environments to be an optional input in the envvar query.

The lagoon-cli has supported this new [query for some time](https://github.com/uselagoon/lagoon-cli/pull/254) so this actual issue shouldn't exist anymore with respect to a role issue. This issue should have been closed when #254 was merged.
